pkgs/games/funkin/default.nix

```diff
@@ -0,0 +1,104 @@
+{ stdenv
+, lib
+, requireFile
+, autoPatchelfHook
+, makeWrapper
+, unzip
+, alsaLib
+, libGL
+, libpulseaudio
+, libX11
+, libXdmcp
+, libXext
+, libXi
+, libXinerama
+, libXrandr
+}:
+let
+  itchioVersion = "4";
+  runtimeDeps = [
+    alsaLib
+    libGL
+    libpulseaudio
+    libX11
+    libXdmcp
+    libXext
+    libXi
+    libXinerama
+    libXrandr
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "funkin";
+  version = "0.2.7";
+
+  # GitHub repo is https://github.com/ninjamuffin99/Funkin/
+  # Building requires Haxe 4.x, HaxeFlixel + abunch of haxelib packages
+  # Waiting for Haxe bump before attempting that
+  # https://github.com/NixOS/nixpkgs/pull/116302
+  src = requireFile rec {
+    name = "${pname}-linux-64bit.zip";
+    url = "https://ninja-muffin24.itch.io/funkin";
+    sha256 = "11a775n5w01sab3ccyqzv03ql01v1x0dv14pf72z1w6ll2ahdnsg";
+    message = ''
+      Please go to ${url}, download version ${itchioVersion} of the 64-bit
+      Linux release (it's free/pay-what-you-want) and add it to the Nix store
+      using either of these options:
+        - nix-store --add-fixed sha256 ${name}
+        - nix-prefetch-url --type sha256 file:///path/to/${name}
+    '';
+  };
+
+  preferLocalBuild = true;
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    unzip
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc
+  ] ++ runtimeDeps;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib/funkin}
+    cp -r * $out/lib/funkin
+    ln -s $out/lib/funkin/Funkin $out/bin/Funkin
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    wrapProgram $out/lib/funkin/Funkin \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath runtimeDeps} \
+      --run "cd $out/lib/funkin"
+  '';
+
+  meta = with lib; {
+    description = "Friday Night Funkin'";
+    longDescription = ''
+      Uh oh! Your tryin to kiss ur hot girlfriend, but her MEAN and EVIL dad is
+      trying to KILL you! He's an ex-rockstar, the only way to get to his heart?
+      The power of music...
+
+      * WASD/ARROW KEYS IS CONTROLS
+      * - and + are volume control
+      * 0 to Mute
+
+      It's basically like DDR, press arrow when arrow over other arrow.
+      And uhhh don't die.
+
+      Originally made for Ludum Dare 47 "Stuck In a Loop".
+    '';
+    homepage = "https://ninja-muffin24.itch.io/funkin";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ OPNA2608 ];
+    platforms = with platforms; windows ++ darwin ++ linux;
+    broken = stdenv.system != "x86_64-linux";
```

Comment:
```suggestion
    platforms = with platforms; [ "x86_64-linux" ];
```

Reply:
I guess that's fine while we're not building from source.

Reply:
I've re-added the original `broken` line because this should be buildable from source on more than just x86_64-linux, but everything else either has binary format or linking problems because of pre-compiled executables and libraries in some Haxelib packages. I've added a comment explaining the situation.